pkg/cluster/config.go

```diff
@@ -0,0 +1,27 @@
+package cluster
+
+import (
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+type Config struct {
+	Cluster               *anywherev1.Cluster
+	VSphereDatacenter     *anywherev1.VSphereDatacenterConfig
+	DockerDatacenter      *anywherev1.DockerDatacenterConfig
+	VSphereMachineConfigs map[string]*anywherev1.VSphereMachineConfig
+	OIDCConfigs           map[string]*anywherev1.OIDCConfig
```

Comment:
why do we need a map for oidc? thought we only allow one

Reply:
This struct tries not to make any assumptions about what is valid and what is not
It simply tries to group everything that can be linked to a `Cluster` object